### PR TITLE
Split and clean up GGUF quantization tests

### PR DESCRIPTION
**Pull request by Isotr0py**
# What does this PR do?

- Split up GGUF tests into three subsets: quantization tests, basic integration tests and models tests.
- This PR also consolidates the quantization tests since they are using duplicate code.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@SunMarc @LysandreJik

### CODE REVIEW

File: tests/quantization/ggml/test_ggml.py [hunk 1]
```diff
@@ -15,27 +15,207 @@
...
+            tokenizer.save_pretrained(tmpdirname)
+
+            model = AutoModelForCausalLM.from_pretrained(tmpdirname).to(torch_device)
+            tokenizer = AutoTokenizer.from_pretrained(tmpdirname)
+
+            text = tokenizer(self.example_text, return_tensors="pt").to(torch_device)
+            out = model.generate(**text, max_new_tokens=10)
+
+        EXPECTED_TEXT = "Hello, World!\n\n[10:0"
+        self.assertEqual(tokenizer.decode(out[0], skip_special_tokens=True), EXPECTED_TEXT)
```
Comment (SunMarc) [inline tests/quantization/ggml/test_ggml.py:196]: maybe we can also check that it matches the output of the model we loaded first ? 
Also, I see that you deleted the other tests, is it intentional ? I'm fine keeping only one gguf format for testing serialization.

**Review by SunMarc — APPROVED** [review]
Nice ! Thanks for the cleanup

File: tests/quantization/ggml/test_ggml.py (hunk 1 above)
Comment (Isotr0py) [reply tests/quantization/ggml/test_ggml.py]: Done! 

I consolidate the quantization tests like `test_q2_k` etc to a parameterized `GgufQuantizationTests`, because their test codes are totally duplicate.

File: tests/quantization/ggml/test_ggml.py [hunk 2]
```diff
@@ -15,27 +15,211 @@
...
+        self.run_gguf_model(gguf_model_id, gguf_filename, expected_text)
+
+    # k-quants
+    @parameterized.expand(
+        [
+            (QuantType.Q2_K.name, "Hello, I'm a 22 year old female"),
+            (QuantType.Q3_K.name, "Hello\n\nI am trying to create a simple program that"),
+            (QuantType.Q4_K.name, "Hello\n\nI am trying to create a simple program that"),
+            (QuantType.Q5_K.name, "Helloveda is a 1999 Indian"),
+            (QuantType.Q6_K.name, "Hello\n\nI am trying to create a simple program that"),
```
Comment (Isotr0py) [inline tests/quantization/ggml/test_ggml.py:76]: We use `GGMLQuantizationType` to make sure it's a validated "pure" quantization type, because some models like `Q4_K_M` means a combination of "mostly Q4_K" with some of other quantization, which shouldn't be used for quantization tests.

BTW, seems that this is causing tests_hub CI failing because gguf is not available, is there any way to solve this?

Comment (SunMarc) [reply tests/quantization/ggml/test_ggml.py:76]: We shouldn't have this issue even if gguf is not available on the hub. There must be an issue when collecting tests with parameterized or maybe an issue with workflow of tests_hub.  cc @ArthurZucker 

Can you try to create a mapping instead so that you can replace GGMLQuantizationType by a string instead ?

Comment (Isotr0py) [reply tests/quantization/ggml/test_ggml.py]: OK, let's use string for parameterize now! Also added a note to keep the quantization style aligned with gguf-py.
